src/spec.c

```diff
@@ -1183,6 +1232,34 @@ int IndexSpec_AddFields(StrongRef spec_ref, IndexSpec *sp, RedisModuleCtx *ctx,
   return rc;
 }
 
+static inline uint64_t HighPart(t_fieldMask mask) { return mask >> 64; }
+static inline uint64_t LowPart(t_fieldMask mask) { return (uint64_t)mask; }
+
+static inline uint16_t TranslateMask(uint64_t maskPart, t_fieldIndex *translationTable, t_fieldIndex *out, uint16_t n, uint8_t offset) {
+  for (int lsbPos = ffsll(maskPart); lsbPos; lsbPos = ffsll(maskPart)) {
+    out[n++] = translationTable[offset + lsbPos - 1];
+    maskPart &= ~(1 << (lsbPos - 1));
+  }
+  return n;
+}
+
+uint16_t IndexSpec_TranslateMaskToFieldIndices(const IndexSpec *sp, t_fieldMask mask, t_fieldIndex *out) {
+  if (!sp->fieldIdToIndex) {
+    return 0;
+  }
+
+  uint16_t count = 0;
+  const uint8_t LOW_OFFSET = 0;
+  if (sizeof(mask) == sizeof(uint64_t)) {
```

Comment:
When is it 64 and when 128?

Reply:
```cpp
#if (defined(__x86_64__) || defined(__aarch64__) || defined(__arm64__)) && !defined(RS_NO_U128)
/* 64 bit architectures use 128 bit field masks and up to 128 fields */
typedef __uint128_t t_fieldMask;
#define RS_FIELDMASK_ALL (((__uint128_t)1 << 127) - (__uint128_t)1 + ((__uint128_t)1 << 127))
#else
/* 32 bit architectures use 64 bits and 64 fields only */
typedef uint64_t t_fieldMask;
#define RS_FIELDMASK_ALL 0xFFFFFFFFFFFFFFFF
#endif
```

Reply:
So the limitation for the number of text fields varies per machine as well??

Reply:
yes, based on the machine architecture.

Reply:
Well, that's bad....
Yet unrelated.